File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/permalink/pathTypes/MonthPathType.java
package com.eden.orchid.posts.permalink.pathTypes;

import com.eden.orchid.api.theme.pages.OrchidPage;
import com.eden.orchid.posts.pages.PostPage;
import com.eden.orchid.posts.permalink.PermalinkPathType;

import javax.inject.Inject;

public class MonthPathType extends PermalinkPathType {

    @Inject
    public MonthPathType() {
        super(100);
    }

    @Override
    public boolean acceptsKey(OrchidPage page, String key) {
        return key.equals("month");
    }

    @Override
    public String format(OrchidPage page, String key) {
        if(page instanceof PostPage) {
            return "" + ((PostPage) page).getMonth();
        }

        return null;
    }

}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/permalink/pathtypes/MonthPathType.kt
package com.eden.orchid.posts.permalink.pathtypes

import com.eden.orchid.api.theme.pages.OrchidPage
import com.eden.orchid.api.theme.permalinks.PermalinkPathType
import com.eden.orchid.posts.pages.PostPage
import javax.inject.Inject

class MonthPathType
@Inject
constructor() : PermalinkPathType(100) {

    override fun acceptsKey(page: OrchidPage, key: String): Boolean {
        return key == "month"
    }

    override fun format(page: OrchidPage, key: String): String? {
        return if (page is PostPage) {
            "${page.month}"
        } else "${page.publishDate.monthValue}"

    }

}

